Server/BackEndMonitoringServer/src/BackEndMonitoringServer.cpp

```diff
@@ -1,27 +1,23 @@
 ﻿#include "stdafx.h"
 
-#include <iostream>
-#include <fstream>
-#include "CAcceptorWrapper.h"
-#include "CThreadPool.h"
-#include "Log.h"
-int main( )
+#include "CService.h"
+#include "CCommandLineHandler.h"
+#include "CServiceHandler.h"
+#include "Utils.h"
+
+int main(int argc, char** argv)
 {
-	std::fstream stream("Log.txt", std::ios_base::out);
-	CLogBuilder builder("Logger", ELogLevel::DEBUG_LEVEL);
-	builder.AddThreadUnsafeStream(stream).SetLogConfig(ELogConfig::CALL_TIME, 
-		ELogConfig::FILE_NAME, ELogConfig::FUNCTION_NAME, 
-		ELogConfig::LINE_NUMBER, ELogConfig::MESSAGE, ELogConfig::PARAMS);
-	auto logger = builder.BuildSharedLog();
-	CEvent event;
-	size_t num_threads = 20;
-	int port = 1111;
-	std::string ip_address = "127.0.0.1";
-	WRITE_DEBUG(*logger, "Start server");
-	CAcceptorWrapper acceptor(port, ip_address, num_threads, event, logger);
+    auto parser = std::make_unique<CommandLineHandler>();
+
+    bool success = parser->Parse(argc, argv);
+
+    if (!success)
+    {
+        Utils::DisplayMessage("Invalid parameters");
+        return EXIT_FAILURE;
+    }
 
-	acceptor.StartServer();
-	system("pause");
-	return 0;
-}
+    const int return_code = success ? 0 : 1;
```

Comment:
If `!success` is `true`, then you step into if statement and `EXIT_FALIURE` will be returned. So you'll never get `return_code` = 1 in this situation. You can safely return `EXIT_SUCCESS`.

Reply:
Thnx, fixed.